### ISSUE TITLE
[Bug] validating integers

### ISSUE THREAD

**Issue by wimby** [moneyphp/money#338]
Hi! I just looked to the source code https://github.com/moneyphp/money/blob/master/src/Number.php#L171 and there seems to be a bug:
```php
function validateNumberAsInteger($number)
{
    // Check if number is invalid because of integer overflow
    $invalid = array_filter(
        str_split($number, strlen((string) PHP_INT_MAX) - 1),
        function ($chunk) {
            // Leading zeros should not invalidate the chunk
            $chunk = ltrim($chunk, '0');
            // Allow chunks containing zeros only
            return '' !== $chunk && false === filter_var($chunk, FILTER_VALIDATE_INT);
        }
    );
    return count($invalid) === 0;
}

assert(!validateNumberAsInteger('123456789012345678-123456'));  // should throw something
```
Can anyone verify and/if fix? Thanks.

**Comment by sagikazarmark** [comment]
@wimby thanks for reporting this issue. I've opened a PR and will release it as 3.1 since there are already additions on this branch.

I also opened a followup ticket (#340) as I spotted another issue as well.

**Comment by frederikbosch** [comment]
Will be merged by #341.
